sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/PayloadSizeGate.java

```diff
@@ -38,12 +38,17 @@
      * @return Buffered data or incoming data depending on threshold condition.
      */
     Flux<ByteBuffer> write(ByteBuffer buf) {
+        /* TODO (gapra): Investigate if there is way to avoid copying the data twice since this may result in lower perf. */
```

Comment:
Can you add that the other copy is when we write it to the pool? Just so we don't forget what we mean later?